src/main/include/utils/CommandUtils.h

```diff
@@ -60,6 +60,38 @@ static frc2::CommandPtr ScoreRamp(std::function<ScoringDirection()> direction,
       .WithTimeout(5_s);
 }
 
+static frc2::CommandPtr PreScoreShuffle(
+    std::function<ScoringDirection()> direction, ScoringSubsystem* scoring,
+    IntakeSubsystem* intake) {
+  return frc2::InstantCommand([] {
+           ConsoleLogger::getInstance().logInfo("Scoring", "Pre Score Shuffle");
+         })
+      .ToPtr()
+      .AndThen((frc2::InstantCommand([scoring, intake, direction] {
+                  scoring->SpinVectorSide(direction());
+                  intake->Out();
+                }).ToPtr())
+                   .Until([intake, direction] {
+                     switch (direction()) {
+                       case ScoringDirection::AmpSide:
+                         return !(intake->NotePresentUpperCenter() ||
```

Comment:
This reads kinda weirdly. Your goal, in English, is "Go until both note present upper and note present <side> are not active", right? Rewrite using boolean AND